repl/src/main/scala/com/cloudera/livy/repl/Main.scala

```diff
@@ -135,17 +133,19 @@ class ScalatraBootstrap extends LifeCycle with Logging {
 
         // Wait for our url to be discovered.
         val replUrl = waitForReplUrl()
-
-        var req = url(callbackUrl).setContentType("application/json", "UTF-8")
-        req = req << write(Map("url" -> replUrl))
-
         info(s"Calling $callbackUrl...")
-        val rep = Http(req OK as.String)
-        rep.onFailure {
-          case _ => System.exit(1)
+        val response = Utils.usingResource(new AsyncHttpClient()) {
+          request => request.preparePost(callbackUrl)
```

Comment:
nit: put `request =>` in the previous line

    val response = Utils.usingResource(new AsyncHttpClient()) { request =>
      // code
    }

Also, `client` is probably more accurate than `request`.